common/src/primitives/amount/rpc.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rpc_description::{HasValueHint, ValueHint as VH};
+
+use super::{Amount, DecimalAmount};
+
+/// Amount type suitable for getting user input supporting both decimal and atom formats in Json.
+#[derive(Clone, Copy, Debug, serde::Serialize, serde::Deserialize, HasValueHint)]
+pub struct RpcAmountIn(RpcAmountData);
+
+#[derive(Clone, Copy, Debug, serde::Serialize, serde::Deserialize, HasValueHint)]
+#[serde(untagged)]
+enum RpcAmountData {
+    Decimal(DecimalAmount),
+    Atoms(Amount),
+}
```

Comment:
If both are provided, it's an error, right?

Reply:
Let me add some tests